agent/handlers/v3/container_metadata_handler.go

```diff
@@ -65,6 +69,7 @@ func ContainerMetadataHandler(state dockerstate.TaskEngineState) func(http.Respo
 func GetContainerResponse(containerID string, state dockerstate.TaskEngineState) (*v2.ContainerResponse, error) {
 	containerResponse, err := v2.NewContainerResponse(containerID, state)
 	if err != nil {
+		seelog.Errorf("unable to get container metadata for container '%s'", containerID)
 		return nil, errors.Errorf("Unable to generate metadata for container '%s'", containerID)
```

Comment:
Should this be TransientError?

Reply:
If it is unable to find container metadata by containerID or TaskID,  I thought it would still be a Bad Request error.

Reply:
> would like to see a unit test case for the transient error.

Added them!

Reply:
Customers interact with this API very generically: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint-v4.html. For v4, they would be curling `${ECS_CONTAINER_METADATA_URI_V4}/task`, the specific task and container IDs are actually generated internally. Since we generate them server-side, if we have an invalid ID in state then that should be a 5xx error.

400 errors should only be used if the customer made a bad request, ie misspelling `task` in the request path.

Reply:
Agreed, will update the PR

Reply:
its kinda both agent/customer.
customers can still specify container ID as part of the request https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint-v2.html

Reply:
That's true, but there are no container Id specific endpoints for v3 and v4
https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint-v3.html
https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint-v4.html